### PR TITLE
minor fixes to cirq intro

### PR DESCRIPTION
**Pull request by balopat**
Minor typos + added some information about `cirq.Simulator.sample` vs `run_sweep` - the exercise right after was a bit confusing without introducing the pandas solution.

### CODE REVIEW

File: docs/tutorials/educators/intro.ipynb [hunk 1]
```diff
@@ -99,18 +99,7 @@
    "metadata": {
```
Comment (rmlarose) [inline docs/tutorials/educators/intro.ipynb:99]: <p>typo: 'nubmer of repetitions' -&gt; 'number of...'</p><p> </p><p>Adding "cirq." would be good for hrefs on the website, e.g. <code>cirq.Sweep</code>,  <code>cirq.Sampler</code>,  etc.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/quantumlib/Cirq/pull/3994/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='118'/>

Comment (rmlarose) [inline docs/tutorials/educators/intro.ipynb:99]: <p>Maybe reword for clarity: "...is the <code>sample</code>  method. This returns a pandas DataFrame where each column is a parameter/measurement key and each row is a repetition."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/quantumlib/Cirq/pull/3994/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='120'/>

Comment (rmlarose) [inline docs/tutorials/educators/intro.ipynb:99]: <p>optional nit: param resolvers -&gt; <code>ParamResolver</code>s</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/quantumlib/Cirq/pull/3994/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='122'/>

Comment (balopat) [reply docs/tutorials/educators/intro.ipynb:99]: ack!

Comment (balopat) [reply docs/tutorials/educators/intro.ipynb:99]: ack!

Comment (balopat) [reply docs/tutorials/educators/intro.ipynb:99]: ack!

**Review by tanujkhattar — APPROVED** [review]
LGTM
